azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java

```diff
@@ -148,6 +156,11 @@ public void updateTrigger(final Trigger t) throws TriggerManagerException {
     this.runnerThread.deleteTrigger(triggerIdMap.get(t.getTriggerId()));
     this.runnerThread.addTrigger(t);
     triggerIdMap.put(t.getTriggerId(), t);
+    if (t.isBackExecuteOnceOnMiss()) {
+      backExecuteEnabledTriggerIds.add(t.getTriggerId());
+    } else {
+      backExecuteEnabledTriggerIds.remove(t.getTriggerId());
```

Comment:
Do you need to run
"if backExecuteEnabledTriggerIds.containsKey(t.getTriggerId())" before removing it?

Reply:
This is a boolean method, if the value is not present, it simply return false but I don't care the return value.

Reply:
Interesting. Thanks for letting me know.